op-e2e/actions/proofs/operator_fee_test.go

```diff
@@ -174,13 +183,67 @@ func Test_ProgramAction_OperatorFeeConsistency(gt *testing.T) {
 
 			receipt, err = env.Alice.GetLastDepositL2Receipt(t)
 			require.NoError(t, err)
-		case NotEnoughFundsInBatch:
+		case NotEnoughFundsInBatch, NotEnoughFundsInBatchMissingOpFee:
+			pkey, err := crypto.GenerateKey()
+			address := crypto.PubkeyToAddress(pkey.PublicKey)
+
+			if testCfg.Custom == NotEnoughFundsInBatchMissingOpFee {
+				// Send `address` just enough ETH to cover the gas costs of the transaction, sans the operator fee.
+				// Since we're just doing a simple call to `Bob`, that should be 21000 gas at the current gas price
+				// plus the L1 data fee.
+
+				// Craft a transaction from Alice -> Bob (just to compute L1 cost, not to send.)
+				env.Alice.L2.ActResetTxOpts(t)
+				env.Alice.L2.ActSetTxToAddr(&env.Dp.Addresses.Bob)(t)
+				tx := env.Alice.L2.MakeTransaction(t)
+
+				rlp, err := tx.MarshalBinary()
+				require.NoError(t, err, "failed to RLP encode transaction")
+
+				unsafeHeader := env.Engine.L2Chain().CurrentHeader()
+				nextBaseFee := eip1559.CalcBaseFee(
+					env.Sd.L2Cfg.Config,
+					unsafeHeader,
+					unsafeHeader.Time+env.Sd.RollupCfg.BlockTime,
+				)
+
+				// Get gas price from oracle
+				gasPriceOracle, err := bindings.NewGasPriceOracleCaller(predeploys.GasPriceOracleAddr, env.Engine.EthClient())
+				require.NoError(t, err)
+
+				daCost := fjordL1Cost(t, gasPriceOracle, types.NewRollupCostData(rlp))
```

Comment:
it may be easier to extract the gas params from the deposit transaction of the block, with `derive.L1BlockInfoFromBytes`

Reply:
Ah neat - removed the GPO method in favor of this.